server/server.go

```diff
@@ -170,10 +163,8 @@ func (srv *Server) Start() error {
 
 // Shutdown shuts down the HTTP, and SSH and health HTTP servers for the Charm Cloud.
 func (srv *Server) Shutdown(ctx context.Context) error {
-	if srv.stats != nil {
-		if err := srv.stats.Shutdown(ctx); err != nil {
-			return err
-		}
+	if err := srv.Config.Stats.Shutdown(ctx); err != nil {
```

Comment:
i think it's good to have the nil check here since `srv.Config.Stats` could be set to nil